contracts/Exclusivity.vy

```diff
@@ -1,90 +1,141 @@
 # @version ^0.3.7
-# this relies on functional ActiveUser contract/class
-# no other contract is interfacing with this code as of 12/12 at 9:00AM
+# relies on only Lottery contract as of June 5 2023
+
 interface ActiveUser:
     def getIsActiveUser(potentialUser: address) -> bool: view
     def getIsAdmin(potentialAdmin: address) -> bool: view
-    
+
+interface Lottery:
+    def getPot() -> uint256: view
+    def setStartingPot(amount: uint256) : nonpayable
+
 activeUserContract: public(ActiveUser)
 
+lotteryContract: public(Lottery)
+
 admin: HashMap[address, bool]
+votingAddress: address
+rickyCWallet: address
 topicsAddress: public(DynArray[address, 1000])
-percentage: uint256
+percentage: uint256 
 classSize: uint256
+activeStudents: HashMap[address, uint256]
+
 @external
-def vote(voter: address):
-    isIn: bool = False
-    for studentAddress in self.topicsAddress: #find index of address of candidate in topics addresses
-            if studentAddress==voter:    
-                isIn = True
-                break
-    if isIn == True:
-        self._removeNonTopics(voter)
-        send(voter,1)
-    if self.admin[voter]:
-     
-        send(voter,(15/100)*self.classSize)
+def __init__(erc20address: address):
+    self.lotteryContract = Lottery(erc20address)
+    self._setRickyC()
+
+# Function 1: students can use Wolvercoin to vote for initiatives; Mr. Theiss's vote can be weighted up to 15% of the total vote:
 
 @external
+def vote(voter: address):
+    if self._isInTopicsList(voter):
+        send(voter, 1)
+    if self._checkAdmin(voter) == True:
+        send(voter, (15/100)*self.classSize)
 
-def tallyVotes(voter: address)-> bool:
+@external
+def tallyVotes(voter: address) -> bool:
     if self.percentage >= 50:
-        self._removeNonTopics(voter)
+        self._removeTopics(voter)
         return True
     return False
 
+# Function 2: Only Honors topics students can have WolverCoin at full functionality
+
+# me when vyper doesnt allow storage variables to have an initial value
+@internal
+def _setRickyC():
+    self.rickyCWallet = 0xF7Edc8FA1eCc32967F827C9043FcAe6ba73afA5c # placeholder
+
+@internal
+def _getRickyC() -> address:
+    return self.rickyCWallet
+
+# uncertain if this belongs in this contract - should eventually be integrated into Token
+@external
```

Comment:
Withdraw needs to use msg.sender as the 'requester' otherwise anyone can withdraw on behalf of someone else.
Is should also assert that the msg.sender is an ActiveUser (replace the _isInTopicsList) call with one to getIsActiveUser from the ActiveUser contract